examples/client.rs

```diff
@@ -22,11 +30,28 @@ fn main() {
 
     let client = Client::new();
 
-    let mut res = client.get(&*url)
-        .header(Connection::close())
-        .send().unwrap();
+    let mut res = match process(&client, &url) {
+        Ok(res) => res,
+        Err(e) => {
+            // this will be things like name server lookup failure
+            println!("Error: {}", e);
+            return;
+        }
+    };
 
+    /*
+     * the Request was handled by the HTTP server.
+     *
+     * Next steps:
+     * Check res.status to find out if it was successful.
+     * Inspect the headers to see what data type was returned and then
+     * parse the response data stored in `contents`.
+     */
     println!("Response: {}", res.status);
     println!("Headers:\n{}", res.headers);
-    io::copy(&mut res, &mut io::stdout()).unwrap();
+
+    let mut contents = String::new();
+    res.read_to_string(&mut contents).unwrap();
```

Comment:
If showing users that we must deal with errors, this seems like a good place to. Running this script against `https://google.com` will trigger this error. For whatever reason, Google doesn't return utf-8 if it doesn't recognize the user-agent.